SwanProjects/bin/swan_kmspecs

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+import argparse
+import json
+import os
+import pprint
+import subprocess
+import sys
+from distutils.spawn import find_executable
+from shutil import rmtree
+
+def _checkipykernel(project_path,python_interpreter,python_version):
+    python_code='import ipykernel'
+    command =  [python_interpreter,"-c",python_code]
+    proc = subprocess.Popen(command, stdout = subprocess.PIPE)
+    proc.wait()
+    data = proc.stdout.read().decode("utf-8")
+    print(data)
+    proc.communicate()
+    if proc.returncode !=0 :
+        print("Error ipykernel not found for {} in project {}".format(python_version,project_path))
+    return proc.returncode
+
+def check_native_kernel(project_path):
+    project_file = project_path+os.path.sep+".swanproject"
+    f = open(project_file,"r+")
+    project_data = f.read() 
+    if project_data.strip() == "":
+        project_data = {}
+    else:
+        project_data = json.loads(project_data)
+    print(project_data)
+    f.seek(0)
+    python2 = find_executable("python2")
```

Comment:
It would be helpful not to leave long pieces of code like this totally uncommented. Please add a comment before every block to explain what is done.

Reply:
documentation added to this script, I know still missing in other places, I am working on it.

fixed